components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/device/UserAuthenticationEndpoint.java

```diff
@@ -68,42 +77,109 @@ public void setDeviceAuthService(DeviceAuthService deviceAuthService) {
     @Consumes("application/x-www-form-urlencoded")
     @Produces("text/html")
     public Response deviceAuthorize(@Context HttpServletRequest request, @Context HttpServletResponse response)
-            throws URISyntaxException, InvalidRequestParentException, IdentityOAuth2Exception, IOException {
+            throws OAuthSystemException {
 
-        String userCode = request.getParameter(Constants.USER_CODE);
-        if (StringUtils.isBlank(userCode)) {
-            if (log.isDebugEnabled()) {
-                log.debug("user_code is missing in the request.");
-            }
-            response.sendRedirect(IdentityUtil.
-                    getServerURL("/authenticationendpoint/device.do?error=invalidRequest", false, false));
-            return null;
-        }
-        String clientId = deviceAuthService.getClientId(userCode);
-        if (StringUtils.isNotBlank(clientId) && StringUtils.equals(getUserCodeStatus(userCode), Constants.PENDING)) {
-
-            setCallbackURI(clientId);
-            deviceAuthService.setAuthenticationStatus(userCode);
-            CommonAuthRequestWrapper commonAuthRequestWrapper = new CommonAuthRequestWrapper(request);
-            commonAuthRequestWrapper.setParameter(Constants.CLIENT_ID, clientId);
-            commonAuthRequestWrapper.setParameter(Constants.RESPONSE_TYPE, Constants.RESPONSE_TYPE_DEVICE);
-            commonAuthRequestWrapper.setParameter(Constants.REDIRECTION_URI, deviceFlowDO.getCallbackUri());
-            if (getScope(userCode) != null) {
-                String scope = String.join(Constants.SEPARATED_WITH_SPACE, getScope(userCode));
-                commonAuthRequestWrapper.setParameter(Constants.SCOPE, scope);
+        try {
+            String userCode = request.getParameter(Constants.USER_CODE);
+            // True when input(user_code) is not REQUIRED.
+            if (StringUtils.isBlank(userCode)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("user_code is missing in the request.");
+                }
+                response.sendRedirect(ServiceURLBuilder.create().addPath(Constants.DEVICE_ENDPOINT_PATH)
+                        .addParameter("error", "invalidRequest").build().getAbsolutePublicURL());
```

Comment:
is `invalidRequest` a standard error parameter?

Reply:
fixed with 'invalid_request'